### DISCUSSION TITLE
Enhancement Request: Real-time Feedback for Large Batch File Processing in Dropzone

### DISCUSSION THREAD

**Discussion by adrydez** [react-dropzone/react-dropzone#1353] [Ideas]
Hello Dropzone Team,

First of all, I would like to express my gratitude for your work on the Dropzone library—it has been an invaluable tool for handling file uploads in our React projects.

However, I've encountered a challenge that I believe could be addressed to improve the library's functionality, especially when dealing with large volumes of files.

Issue Description:
When users drop a large number of files (e.g., thousands) into Dropzone, there is currently no built-in way to provide them with real-time feedback on the processing status of these files. The **onDrop** method gets triggered only after all files have been processed, which can lead to a significant delay without any feedback for the user, making the interface feel unresponsive.

I attempted to use the **validator** function to implement custom logic for real-time feedback, hoping to update a counter or a progress bar as each file is validated. Unfortunately, it appears that updates made within the **validator** function do not reflect in real-time in the UI. For example, attempting to update a React state or manipulating the DOM directly within this function results in the changes being batched and only displayed after all files have been processed, not incrementally.

Suggested Enhancement:
It would be incredibly beneficial for Dropzone to support or offer a pattern for implementing real-time feedback during the file processing phase, especially when handling a large batch of files. An enhancement to the **validator** function or a new callback/event that fires per file (or batch of files) during processing could potentially address this issue.

Potential Benefits:
Improved User Experience: Users get immediate feedback, improving the perceived responsiveness of the application.
Enhanced Usability for Large File Batches: Makes Dropzone more suitable for applications that require handling of large numbers of files, enhancing its versatility.
Conclusion:
The ability to provide real-time feedback during the processing of a large number of files would be a significant enhancement to Dropzone, aligning with the needs of applications that manage extensive datasets or file collections.

Thank you for considering this enhancement request. I am looking forward to any suggestions or discussions on how we could address this challenge.

Best regards,
AdRY

**Comment by adrydez** [comment]
just to add more information, I know, is not usual to work with thousands of files but it's real to work with vpn connections. And often there are lot of delays and you don't know if even a single file has been accepted.
So, to have a visual feedback that something is happening it would be great

**Comment by rolandjitsu** [comment]
@adrydez thank you for your feedback.

I do see value in what you're suggesting, unfortunately, I think there may be a browser limitation that will not allow processing n files at a time. We use https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/drop_event, so if that event takes some time to process, there's nothing we can do on our side. However, the `getFilesFromEvent` is a function we control, so that could be fine tuned to provide some kind of progress or cb every n files.

Would you be open to a PR? I don't have the luxury of time at the moment to impl. such a feature.

**Comment by t634146397** [comment]
hi, how was it ultimately resolved?

**Reply by rolandjitsu** [reply]
It wasn't. It would probably require quite some work.
